mantle/platform/api/aws/images.go

```diff
@@ -338,6 +340,78 @@ func (a *API) CreateImportRole(bucket string) error {
 	return nil
 }
 
+func (a *API) CreateWinLiImage(snapshotID string, name string, description string, architecture string, windowsAMI string, instanceType string, volumetype string) (string, string, error) {
+	//var awsArch string
+	if architecture != "x86_64" {
+		return "", "", fmt.Errorf("unsupported WinLi architecture %q", architecture)
+	}
+
+	opts := *a.opts
+	opts.AMI = windowsAMI
+	opts.InstanceType = instanceType
+	opts.SecurityGroup = "winli-builder"
+	aa, err := New(&opts)
+	if err != nil {
+		return "", "", err
+	}
+
+	id := strings.Split(uuid.New().String(), "-")[0]
+	instanceName := "winli-builder-" + id
```

Comment:
Thanks for doing this. one further idea would be to take `randomName()` from [azure api code](https://github.com/coreos/coreos-assembler/blob/d31cd4cb3eec267806338e0231bcc64949971cd2/mantle/platform/api/azure/api.go#L120-L126) and put it in util so everyone could use it.